xrmocap/utils/eval_utils.py

```diff
@@ -137,13 +161,28 @@ def align_convention_mask(pred_keypoints3d_raw: Keypoints,
     else:
         gt_keypoints3d = gt_keypoints3d_raw
 
-    if pred_kps3d_convention != gt_kps3d_convention:
-        if eval_kps3d_convention != 'human_data':
-            logger.warning(
-                'Predicion and ground truth is having'
-                'different convention. It is recommended to set '
-                'eval_kps3d_convention to human_data to avoid error.')
+    pred_kps3d_mask = pred_keypoints3d.get_mask()
+    pred_kps3d = pred_keypoints3d.get_keypoints()[..., :3]
+    if pred_nose is not None and eval_kps3d_convention == 'campus':
+        pred_kps3d = add_campus_jaw_headtop(pred_nose, pred_kps3d)
```

Comment:
I suggest keeping this function simple and doing the extra keypoints approximation and addition for gt and pred outside this function if it is necessary for certain methods and conventions during evaluation. Since some methods might need simple convention alignment only but do not need keypoints approximation.
This function aims to align `pred convention` and `gt convention` to the same `eval convention` since they might differ when using different 2D perception models. The eval convention can be a new convention such as `human_data`(recommended) or one of the gt convention and eval convention. This function also masks out the extra keypoints that only exist in pred or gt since they cannot be evaluated properly.
For example, if head and jaw keypoints are available in gt but not available in pred, this function will mask out the head and jaw keypoints in gt and skip the further evaluation for these two keypoints.